### ISSUE TITLE
Toggles: show a popover with an explanation for disabled toggles

### ISSUE THREAD

**Issue by rickybanister** [Automattic/wp-calypso#14059]
@simonwheatley mentioned in p4k3M4-2kR-p2 that disabled toggles can be confusing because they aren't visually differentiated enough from deactivated toggles.

Let's provide an option to display a popover with a message (on a per-setting basis) explaining why an option and its toggle are disabled. 

Example:

![image](https://cloud.githubusercontent.com/assets/437258/26070740/b29a7e4e-3973-11e7-8eb4-7edcc1c4c3e4.png)

cc @lancewillett

**Comment by lancewillett** [comment]
> Confused because there are two shades for the same state (off and enabled, vs off and disabled)

Two other ideas are:

1. Hide the sub-toggles until the parent toggle is turned on.
2. Differentiate the main (parent) toggle on/off state with something more than color saturation. Say with a line or dot like iOS does in accessibility mode (on even `on` and `off`, but that's hard with translations).

**Comment by simonwheatley** [comment]
> Let's provide an option to display a popover with a message (on a per-setting basis) explaining why an option and its toggle are disabled.

I think adding a popover would increase the concentration needed for me to understand the state of the toggles at a glance. Just my 2p.

**Comment by rickybanister** [comment]
So to take a step back, we decided to show the sub-options because it made the feature more 'discoverable' and let users understand what exactly turning on the main toggle would actually enable. 

We've 'waffled' a bit back and forth about this since—

* Should we allow the sub-toggles to toggle the main toggle? We decided that was confusing. When interacting with things you could accidentally enable something else you did not want within the feature.
* Hiding the sub-options and the preview made the feature (in this case related posts) more opaque.
* Adding the 'on/off' indicators like Apple does doesn't seem to help, I find those confusing and never know which means 'on'. Also that would not help indicate in this case which option is disabled. 

@simonwheatley can you explain why the pop-over would increase your concentration? I meant to imply that the popover would only show if you tried to activate the disabled toggle. If calypso could play sounds it would honk a horn at you to tell you 'you can't do that, here is why'. Does that make sense?

**Comment by simonwheatley** [comment]
> can you explain why the pop-over would increase your concentration?

I meant that if I cannot understand what a UI element does by looking at it, then adding the popover interaction adds the following actions and mental steps:

1. Wonder what the toggle state is
2. Guess that I can move my mouse and get a popover
3. See and read the popover
4. Refactor my mental model of what's happening on screen
5. Decide what to do

Maybe I'm being too curmudgeonly 😬 I think because I expect a toggle to have two state (on / off) and I saw two states represented visually (off and enabled / off and disabled), for some reason that confusion short-circuited my ability to understand what I could do on this screen.

**Comment by rickybanister** [comment]
Thanks for clarifying. That does make sense, but I think it's just a matter of circumstances here. I'd like to do what we can to improve the clarity of this screen, but we aren't doing anything different than say if the 'sub-toggles' were checkboxes. A checkbox form element can have both an 'unchecked' state and a 'disabled' state, we're basically doing the same—our toggles are just not as ubiquitous as a checkbox or a radio button.

**Comment by rickybanister** [comment]
@ebinnion this is tangentially related to the work you've been doing with the module list. I know that work won't resolve this issue, but worth keeping this one open as they are related.
